source/agora/consensus/ValidatorSet.d

```diff
@@ -319,6 +320,35 @@ public class ValidatorSet
         }
     }
 
+    /***************************************************************************
+
+        Extracts the UTXO hash value of a valid validators at a next
+        block height.
+
+        Params:
+            block_height = the next block height
+            utxo_key_set = Store the validators's UTXO hash value.
+
+    ***************************************************************************/
+
+    public void getAvailableKeys (ulong block_height, ref Set!Hash utxo_key_set) @safe nothrow
```

Comment:
The `utxo_key_set` seems unnecessary. This function could be renamed `ulong getValidatorCount (ulong block_height)`. Then all you have to do is call:

```D
this.db.execute("SELECT count(*) FROM validator_set WHERE enrolled_height >= ?", height).oneValue!ulong;
```

Reply:
Shouldn't there be more than X verifiers?
If only one validator is needed, I agree with you. 
However, When we need several, shouldn't we remove the same one and check the number?

Reply:
`SELECT count(*)` returns the count.